source/arm7/tmio.twl.c

```diff
@@ -18,7 +18,7 @@
 #define INIT_DELAY_FUNC()  swiDelay(TMIO_CLK2DIV(400000u) * 74 / 4)
 
 
-static u32 g_status[2] = {0};
+static _Atomic u32 g_status[2] = {0};
```

Comment:
Why is this needed?

Reply:
The C11 standard requires variables to be marked as _Atomic in order to use atomic operations. GCC seems to have a "bug" of sorts that allows this to compile without marking it properly.

Reply:
Sure, but what's the issue if this isn't marked as atomic?

Reply:
statusPtr uses atomic operations so LLVM, following the standard, must have _Atomic on it. g_status needs it because statusPtr is a pointer to it, and _Atomic pointers are not compatible with the regular equivalent in GCC. Perhaps this could be solved with a cast, although I am unsure whether or not this could cause issues.
```c
source/arm7/tmio.twl.c: In function 'TMIO_sendCommand':
source/arm7/tmio.twl.c:254:36: error: initialization of '_Atomic u32 * const' {aka _Atomic long unsigned int * const'} from incompatible pointer type 'u32 *' {aka 'long unsigned int *'} [-Wincompatible-pointer-types]
  254 | 32 *const statusPtr = &g_status[controller];
      |                       ^
```

Reply:
In that case, you need to describe the problem a bit in the PR description.

Reply:
I clarified my comment & updated the description to reflect the current state of the PR.

Reply:
Okay, so this is causing build issues because of missing atomic builtins. It just won't build in GCC.

I think the right thing to do would be to get rid of all the atomic constructs and switch to using `volatile` if needed.

Reply:
Latest commit should fix the linking errors; the SDK, examples, and templates now build on my machine with GCC, and libnds compiles with LLVM.